src/main/scala/org/broadinstitute/dsde/firecloud/utils/RestJsonClient.scala

```diff
@@ -49,33 +51,49 @@ trait RestJsonClient extends FireCloudRequestBuilding {
     }
 
     val finalPipeline = addCreds.map(creds => creds ~> pipeline).getOrElse(pipeline)
-    finalPipeline(req)
+    finalPipeline(req) map { res =>
+      if (label.nonEmpty && tick != -1) {
+        val elapsed = System.currentTimeMillis() - tick
+        perfLogger.info(perfmsg(label.get, elapsed))
+      }
+
+      res
+    }
   }
```

Comment:
It looks like the logging code could be factored out since it's already used in more than one place, via a utility method in `PerformanceLogging`, like so:

```scala
def withPerfLogging(body: => Any): Unit = {
    val tick = System.currentTimeMillis()

    body

    val elapsed = System.currentTimeMillis() - tick

    perfLogger.info(...)
  }
```

which can be used by clients via

```scala
withPerfLoging {
    addCreds.map(creds => creds ~> pipeline).getOrElse(pipeline)
    .
    .
    .
  }
```

Reply:
I thought about this, but it's particularly awkward in the current code, and I really wanted to make this quick and dirty (albeit not introducing tech debt or instability). One instance of the perf timing is split across the `requestToObject` and `resultsToObject` methods and so wouldn't fit inside a closure. The other instance, inside `doRequest`, is also split across a closure boundary - it starts where the pipeline (Future) is created, and ends inside the `map` to read the results. So it would require a nontrivial amount of refactoring.